### PR TITLE
Decrease `request_timeout` in builders to avoid panics

### PR DESCRIPTION
**Pull request by elmattic**
## Summary of changes

Changes introduced in this pull request:

- Decrease `request_timeout` value in `WsClientBuilder` and `HttpClientBuilder` to avoid panics (see logs below)

Rational:

We don't need to handle timeouts at the `jsonrpsee` client level since we shim our own timeouts during the call, but setting it to some max value seems unsafe.

Setting a 10s timeout and simulating a slow endpoint in forest daemon:

```console
$ FOREST_RPC_DEFAULT_TIMEOUT=10 ./forest-tool api compare --lotus /ip4/127.0.0.1/tcp/1234/ws --forest /ip4/127.0.0.1/tcp/2345/ws forest_snapshot_calibnet_2024-04-19_height_1539740.forest.car.zst --filter EthChainId
Request dump: RpcRequest { method_name: "Filecoin.EthChainId", params: Null, result_type: PhantomData<serde_json::value::Value>, api_version: V1, timeout: 10s }
Request params JSON: null
Lotus response: "0x4cb2f"

| RPC Method          | Forest              | Lotus |
|---------------------|---------------------|-------|
| Filecoin.EthChainId | InternalServerError | Valid |
Error: Some tests failed
```

## Reference issue to close (if applicable)

Closes

## Other information and links

```
$ RUST_BACKTRACE=1 ./forest-tool api compare --lotus /ip4/127.0.0.1/tcp/1234/ws --forest /ip4/127.0.0.1/tcp/2345/ws forest_snapshot_calibnet_2024-04-19_height_1539740.forest.car.zst --filter EthChainId
thread 'tokio-runtime-worker' panicked at library/std/src/time.rs:417:33:
overflow when adding duration to instant
```

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

**Comment by elmattic** [comment]
> looks good.
> 
> nit: I'm still unsure what does `// we handle timeouts ourselves` comment mean, but it's perhaps out of scope anyway.

Check this code: https://github.com/ChainSafe/forest/blob/main/src/rpc/client.rs#L65-L97

File: src/rpc/client.rs [hunk 1]
```diff
@@ -153,7 +153,7 @@ impl Debug for UrlClient {
 
 impl UrlClient {
     async fn new(url: Url, token: impl Into<Option<String>>) -> Result<Self, ClientError> {
-        let timeout = Duration::MAX; // we handle timeouts ourselves.
```
Comment (LesnyRumcajs) [inline src/rpc/client.rs:156]: Will a subscription method like `ChainNotify` behave correctly with this change?

Comment (elmattic) [reply src/rpc/client.rs:156]: Right now, we don't have a clear plan for testing such methods, so we ignore it.

Comment (LesnyRumcajs) [reply src/rpc/client.rs:156]: Why do we ignore it? I understand we don't have an automated way of testing this yet. If you set locally the timeout to 10s, will this method still work as expected after 15s?

Comment (LesnyRumcajs) [reply src/rpc/client.rs:156]: Also, why does the overflow occur? Why is a value added to the timeout we don't use?

Comment (elmattic) [reply src/rpc/client.rs:156]: > Why do we ignore it?

Testing a method like `ChainNotify` and `EthSubscribe` requires new infrastructure code that we don't have at the moment.

Comment (elmattic) [reply src/rpc/client.rs:156]: > If you set locally the timeout to 10s, will this method still work as expected after 15s?

Which method are you referring to?

**Comment by hanabi1224** [comment]
> overflow when adding duration to instant

This seems like a bug in upstream lib, should we PR a fix as well?

File: src/rpc/client.rs (hunk 1 above)
Comment (LesnyRumcajs) [reply src/rpc/client.rs:156]: I'm referring to `ChainNotify`. We implemented it, so we had to test it somehow, right?

**Review by LesnyRumcajs — APPROVED** [review]

**Pull request merged**